resources/assets/js/components/ActivityFeed.tsx

```diff
@@ -1,182 +1,133 @@
-/* eslint-disable camelcase */
-/* eslint-disable @typescript-eslint/camelcase */
-import React, { useEffect, useState } from "react";
+import * as React from "react";
 import { connect } from "react-redux";
-import { FormattedMessage, useIntl } from "react-intl";
-import Activity from "./Activity";
+import {
+  useIntl,
+  FormattedMessage,
+  MessageDescriptor,
+  defineMessages,
+} from "react-intl";
 import { RootState } from "../store/store";
-import { Comment } from "../models/types";
-import { DispatchType } from "../configureStore";
-import { fetchComments } from "../store/Job/jobActions";
-import { getComments, sortComments } from "../store/Job/jobSelector";
-import { commentTypeMessages } from "./CommentForm";
-import { activityLocationOption } from "../models/localizedConstants";
-import { activityLocationUrl } from "../models/jobUtil";
+import { getComments } from "../store/Job/jobSelector";
+import { hasKey } from "../helpers/queries";
 import { LocationId } from "../models/lookupConstants";
+import CommentForm from "./CommentForm";
+import ActivityList from "./ActivityList";
+import Icon from "./Icon";
+
+const messages = defineMessages({
+  loadingIcon: {
+    id: "activityfeed.loadingIconText",
+    defaultMessage: "Number of activities is loading...",
+    description: "Accessible text for the loading icon",
+  },
+});
 
 interface ActivityFeedProps {
   jobId: number;
   isHrAdvisor: boolean;
-  comments: Comment[];
-  handleFetchComments: (jobId: number) => Promise<void>;
-  filterComments?: (comment: Comment) => boolean;
+  generalLocation: string;
+  locationMessages: { [LocationId: string]: MessageDescriptor };
+  totalActivities: number;
```

Comment:
I think you should change the interface slightly. ActivityFeed should accept a list of comments, count them inside the component instead of recieving the count as a prop, and pass the comments to ActivityList. The exported, redux-connected component should take an optional filterComments prop, just like ActivityList.

a) This would make it impossible for this component to be given a count which doesn't accurately reflect what's shown in ActivityList.
b) If other components are including ActivityFeed instead of ActivityList, they need a way to filter the comments.

Reply:
Okay will do. For a) though, the count would should always be accurate since it's being grabbed from the root state.